pkg/tsdb/loki/frame.go

```diff
@@ -209,3 +216,75 @@ func getFrameLabels(frame *data.Frame) map[string]string {
 
 	return labels
 }
+
+func parseStats(frameMetaCustom interface{}) []data.QueryStat {
```

Comment:
this was moved here from `parse_response.go`, and adjusted for the different types.

in a future PR we can switch this to a simpler approach like this : https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/result_transformer.ts#L304 , but for now i wanted to keep the diff small.